addons/mail/models/mail_activity.py

```diff
@@ -77,12 +83,29 @@ def default_get(self, fields):
         ('today', 'Today'),
         ('planned', 'Planned')], 'State',
         compute='_compute_state')
+    next_activity_type_id = fields.Many2one("mail.activity.type", string="Recommended Activities")
+    previous_activity_type_id = fields.Many2one('mail.activity.type', 'Previous Activity', default=False)
+    # To display or hide recommended activity label
+    use_next_activity = fields.Boolean(compute='_is_next_activity_enable')
+
+    @api.multi
+    @api.onchange('previous_activity_type_id')
+    def _is_next_activity_enable(self):
+        for record in self:
+            record.use_next_activity = True if record.previous_activity_type_id and record.previous_activity_type_id.next_type_ids else False
 
     @api.depends('res_model', 'res_id')
     def _compute_res_name(self):
         for activity in self:
             activity.res_name = self.env[activity.res_model].browse(activity.res_id).name_get()[0][1]
 
+    @api.onchange('previous_activity_type_id')
+    def _onchange_previous_activity_type_id(self):
+        for act in self:
+            next_acts = act.previous_activity_type_id and act.previous_activity_type_id.next_type_ids.sorted(key=lambda r: r.id)
```

Comment:
activity.next_activity_id = activity.previous_activity_type_id.next_type_ids.sorted().id

Is the sorted necessary actually ?